pkg/syncer/volume_health_reconciler.go

```diff
@@ -91,16 +143,63 @@ func NewVolumeHealthReconciler(
 		svcPVCSynced:        svcPVCInformer.Informer().HasSynced,
 		svcClaimQueue:       svcClaimQueue,
 		supervisorNamespace: supervisorNamespace,
+		volumeHandleToPVs: &volumeHandleToPVs{
+			RWMutex: &sync.RWMutex{},
+			items:   make(map[string][]string),
+		},
 	}
 
 	svcPVCInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
 		AddFunc:    rc.svcAddPVC,
 		UpdateFunc: rc.svcUpdatePVC,
 	}, resyncPeriod)
 
+	tkgPVInformer.Informer().AddEventHandlerWithResyncPeriod(cache.ResourceEventHandlerFuncs{
+		AddFunc:    rc.tkgAddPV,
+		UpdateFunc: rc.tkgUpdatePV,
+		DeleteFunc: rc.tkgDeletePV,
+	}, resyncPeriod)
+
 	return rc
 }
 
+func (rc *volumeHealthReconciler) tkgAddPV(obj interface{}) {
+	pv, ok := obj.(*v1.PersistentVolume)
+	if !ok {
+		return
+	}
+	if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == csitypes.Name && pv.Status.Phase == v1.VolumeBound {
```

Comment:
For Statically Crated PV, they will be added with the status Pending and then go to the available Phase.

Reply:
We only need this mapping when the PV is bound to a PVC. This mapping is used to add the volume health annotation to that PVC..